File: past-competitions/2018/src/brett-cpp/Prob02.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[])
{
	// open file for reading
	FILE *inputFile = fopen("Prob02.in.txt", "r");
	int testCases;

	// read number of test cases
	fscanf(inputFile, "%d\n", &testCases);
	
	int vowelCount = 0;

	// execute test cases
	while(testCases > 0){
		// reset vowel count
		vowelCount = 0;

		// get the next character
		char letter = fgetc(inputFile);
		// until we hit a newline
		while(letter != '\n'){
			// count the vowels - a, e, i, o, u
			if(letter == 'a' || letter == 'e' || letter == 'i' || letter == 'o' || letter == 'u'){
				vowelCount++;
			}
			// get the next character
			letter = fgetc(inputFile);
		}

		// print the count
		printf("%d\n", vowelCount);

		testCases = testCases - 1;
	}

	// close the file
	fclose(inputFile);
}

File: self-solutions/2018/Prob02.py
with open("Prob02.in.txt", "rt") as inputFile:
    cases = int(inputFile.readline().replace("\n", ""))
    for caseNum in range(cases):
        line = inputFile.readline().replace("\n", "")
        count = 0
        for i in range(len(line)):
            if line[i] == 'a' or line[i] == 'e' or line[i] == 'i' or line[i] == 'o' or line[i] == 'u':
                count = count + 1
        print(count)
